src/main/java/tasks/Task.java

```diff
@@ -0,0 +1,88 @@
+package tasks;
+
+import java.time.LocalDate;
+
+/**
+ * The Task class represents a general task with a description and a completion status.
+ * This class is intended to be extended by more specific task types.
+ */
+
+public class Task {
+    protected String action;
```

Comment:
Perhaps consider changing the `action` variable name to `description` to match the JavaDoc header?